server/src/main/java/org/elasticsearch/index/search/QueryParserHelper.java

```diff
@@ -120,6 +120,9 @@ static Map<String, Float> resolveMappingField(
         String fieldSuffix
     ) {
         Set<String> allFields = context.getMatchingFieldNames(fieldOrPattern);
+        // avoid failing search request when using the default value for index.query.default_field (*)
+        int limit = Regex.isMatchAllPattern(fieldOrPattern) ? IndexSearcher.getMaxClauseCount() : Integer.MAX_VALUE;
```

Comment:
I'm tempted to say we should always apply this limit also if someone uses `labels.*`. If I search for `*` things work, but if I make it more specific to `labels.*` the query might break. This sounds surprising. What is the use case it is beneficial for the user that the query would fail completely?

Reply:
This is obviously something that needs a debate. But my reasoning was that by selecting all fields with `*`, you're not interested in any one specific field or namespace but rather want to query as many fields as possible without failing. But when you define a specific namespace such `labels.*`, you might have the expectation that this actually searches in all labels fields and silently ignoring fields may be more surprising compared to truncating searched fields when you're not specific about a namespace.

This also begs the question what happens if you define multiple patterns in the fields to query, such as `labels.*` and `annotations.*`. What if `labels.*` already contains more fields than the limit? Would we silently ignore `annotations.*`?

Just applying the truncation logic to `*` sidesteps that question.

Another option would be to have a flag to opt-out of the truncation or return some kind of warning to the client. But both of these options are much more complex to implement.

> If I search for *

Maybe it's just wording but this is not about what you search for but which fields you're searching. So this applies to `default_field` as well as the [`multi_match_query`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html), for example.

Reply:
++ on side stepping this discussion for now and wait for user feedback on it. We can always follow up.